File: src/ssr_pkg/scripts/chao_node_1.py
#!/usr/bin/env python3 
#coding=utf-8
 
import rospy
from std_msgs.msg import String
from qq_msgs.msg import Carry
if __name__=="__main__":
    rospy.init_node("chao_node")
    rospy.logwarn("chao_node is started ")

    pub=rospy.Publisher("chao_node_toptic",Carry,queue_size=10)

    rate=rospy.Rate(1)

    while not rospy.is_shutdown():
        rospy.loginfo("chao_node is running")
        msg=Carry()
        msg.grade="王者"
        msg.star=50
        msg.data="This is chao_node publishing data\n"
        pub.publish(msg)
        rate.sleep()

File: src/ssr_pkg/src/chao_node_1.cpp
//使用新定义的消息类型
#include<ros/ros.h>
#include<std_msgs/String.h>
#include<qq_msgs/Carry.h>

int main(int argc,char  *argv[])
{
    ros::init(argc,argv,"chao");
    printf( "My Gun Go And back\n");
    ros::NodeHandle nh;
    ros::Publisher pub=nh.advertise<qq_msgs::Carry>("kuaishangchekaiheiqun",10);
    ros::Rate loop_rate(10);//频率控制对象
    while (ros::ok())
    {
        printf("chao_刷屏\n");
        qq_msgs::Carry msg;
        msg.grade="王者";
        msg.star=100;
        msg.data="马超带你飞\n";
        pub.publish(msg);
        loop_rate.sleep();//调用对象
    }
    
    return 0;
}